source/ConditionsStore.cpp

```diff
@@ -475,3 +496,26 @@ const ConditionsStore::ConditionEntry *ConditionsStore::GetEntry(const string &n
 	// And otherwise we don't have a match.
 	return nullptr;
 }
+
+
+
+// Helper function to check if we can safely add a provider with the given name.
+bool ConditionsStore::VerifyProviderLocation(const string &name) const
+{
+	auto it = storage.upper_bound(name);
+	if(it == storage.begin())
+		return true;
+
+	--it;
+	// If the entry is already there, then it apparently was safe to add the entry.
+	if(name == it->first)
+		return true;
+
+	const ConditionEntry &ce = it->second;
+	if(ce.provider && ce.provider->isPrefixProvider && 0 == name.compare(0, ce.provider->name.length(), ce.provider->name))
+	{
+		Logger::LogError("Error: not adding provider for \"" + name + "\", because it is within range of prefixed derived provider \"" + ce.provider->name + "\".");
```

Comment:
```suggestion
		Logger::LogError("Error: not adding provider for \"" + name + "\""
				", because it is within range of prefixed derived provider \"" + ce.provider->name + "\".");
```